linked_list/implementation.py

```diff
@@ -7,34 +8,151 @@ class LinkedList(AbstractLinkedList):
     """
 
     def __init__(self, elements=None):
-        pass
+        
+        self.elements = elements
```

Comment:
this is a python container which you call on your `__add__` method although you do not update it on `append` or on `pop`. Also it is probably not in the spirit of the assignment to keep an instance level variable with a list of elements.